temba/channels/models.py

```diff
@@ -323,6 +328,25 @@ def add_viber_channel(cls, org, user, name):
         return Channel.create(org, user, None, Channel.TYPE_VIBER, name=name, address=Channel.VIBER_NO_SERVICE_ID)
 
     @classmethod
+    def add_fcm_channel(cls, org, user, data):
+        """
+        Creates a new Firebase Cloud Messaging channel
+        """
+        from temba.contacts.models import FCM_SCHEME
+
+        assert Channel.CONFIG_FCM_KEY in data and Channel.CONFIG_FCM_TITLE in data, "%s and %s are required" % (
+            Channel.CONFIG_FCM_KEY, Channel.CONFIG_FCM_TITLE)
+
+        existing = Channel.objects.filter(is_active=True, org=org, channel_type=Channel.TYPE_FCM).first()
+        if existing:
+            existing.config = json.dumps(data)
+            existing.save(update_fields=('config',))
+            return existing
+        else:
+            return Channel.create(org, user, None, Channel.TYPE_FCM, name=org.name, address="fcm-%s" % org.slug,
```

Comment:
Is there any kind of validation we can do of the tokens they have handed us? Always nice to fail fast when the user is inputting stuff to make sure they put in the right tokens.